src/lib/operators/insert.cpp

```diff
@@ -231,13 +239,17 @@ void Insert::_on_commit_records(const CommitID cid) {
 
     // This fence ensures that the changes to TID (which are not sequentially consistent) are visible to other threads.
     std::atomic_thread_fence(std::memory_order_release);
+
+    // Deregister the pending Insert and try to finalize the chunk.
+    mvcc_data->deregister_insert();
+    target_chunk->try_finalize();
```

Comment:
I'll add a comment, it might just happen that we're the last pending Insert operator for an already full chunk. Thus, we try to finalize it (which will do nothing if the chunk is not full).

Reply:
```c++
// Deregister the pending Insert and try to finalize the chunk. We might be the last committing Insert operator
// inserting into a chunk that reached its target size. In this case, the Insert operator that added a new chunk to
// the table allowed the chunk to be finalized and `try_finalize()` actually does that. Otherwise, this is a no-op.
```

Reply:
"allowed the chunk to be finalized" ... mention setting the `is_finalizable` flag? Would have helped me.

It is not clear what "and `try_finalize()` actually does that" refers to. On allowing the chunk to be finalized?

Reply:
like that?
```c++
// Deregister the pending Insert and try to finalize the chunk. We might be the last committing Insert operator
// inserting into a chunk that reached its target size. In this case, the Insert operator that added a new chunk to
// the table allowed the chunk to be finalized, i.e., it set the _is_finalizable flag. Then, `try_finalize()`
// actually finalizes the chunk. Otherwise, this is a no-op.
```

Reply:
Ack.

`_is_finalizable` in "``"?